### PR TITLE
[AMORO-3235] Support action filter by ams backend

### LINKED ISSUE #3235: [Subtask]: Support action filter by ams backend
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

The interface is:
/optimize/actions

For now, it will return:
[minor, major, full, committing, planning, pending, idle]
which is referenced to OptimizingStatus enum

### Parent issue

#3139

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by majin1102**
## Why are the changes needed?

Close #3235.

## Brief change log

- Support action filter for optimizing page
- The front work is : https://github.com/apache/amoro/pull/3201

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by zhoujinsong — APPROVED** [review]
LGTM.

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/OptimizerGroupController.java [hunk 1]
```diff
@@ -58,13 +62,24 @@ public OptimizerGroupController(
...
+                .map(OptimizingStatus::displayValue)
+                .collect(Collectors.toList())));
+  }
+
   /** Get optimize tables. * @return List of {@link TableOptimizingInfo} */
   public void getOptimizerTables(Context ctx) {
     String optimizerGroup = ctx.pathParam("optimizerGroup");
     String dbFilterStr = ctx.queryParam("dbSearchInput");
     String tableFilterStr = ctx.queryParam("tableSearchInput");
+    String[] requestedActions = ctx.queryParamAsClass("actions", String[].class).getOrDefault(null);
```
Comment (zhoujinsong) [inline amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/OptimizerGroupController.java:78]: I got following information when trying to ran this PR in my local environment, It seems Javalin cannot parse string array param by default.

```
[JettyServerThreadPool-44] [io.javalin.Javalin] [] - Can't convert to String[]. Register a converter using JavalinValidation#register.
```

**Review by zhoujinsong — CHANGES_REQUESTED** [review]
@majin1102 I got some error when trying to validate this PR in my local environment.

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/OptimizerGroupController.java (hunk 1 above)
Comment (majin1102) [reply amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/OptimizerGroupController.java]: Sorry for that.

I have tested the new code, PTAL

**Comment by majin1102** [comment]
> @majin1102 I got some error when trying to validate this PR in my local environment.

To be reviewed

**Review by baiyangtx — APPROVED** [review]

**Review by zhoujinsong — APPROVED** [review]
Validated successfully in my local environment.

**Pull request merged**